src/mame/drivers/viper.cpp

```diff
@@ -75,27 +82,44 @@
 
     TODO:
     - needs a proper way to dump security dongles, anything but p9112 has placeholder ROM for ds2430.
+    - figure out why games randomly crash (IRQ related?)
+
+    Other notes:
+    - "Distribution error" means there's a region mismatch.
+    - Games that hang randomly seem to hang on IRQ16 possibly? You can see "IRQ16 taken" but it hangs before you see "IRQ16 cleared".
+    - Hold TEST while booting (from the very start) to initialize the RTC for most games.
+    - It seems that p911 has 3 unique regional images: U/E, K/A, and J. If you try booting, for example, U region on a K/A image, it won't find some files and will error out with "distribution error".
 
     Game status:
-        ppp2nd              Boots in-game, no sound or DVD support, crashes when going into song selection screen
-        boxingm             Goes to attract mode when ran with memory card check. Coins up.
-        code1d,b            RTC self check bad
-        gticlub2,ea         Attract mode works. Coins up. Hangs in car selection.
-        jpark3              POST?: Shows "Now loading..." then black screen (sets global timer 1 on EPIC) - with IRQ3 crashes at first 3d frame
-        mocapglf            Sensor I/O error
-        mocapb,j            Crash after self checks
-        p911                "Distribution error"
-        p911e,j,uc,kc       Hangs at POST, with IRQ3 it crashes at first 3d frame
-        p9112               RTC self check bad
+        boxingm             Goes in-game. Controllers are not emulated. Various graphical glitches.
+        jpark3              Goes in-game. Controllers are not emulated. Various graphical glitches.
+        mocapb,j            Goes in-game. Controllers are not emulated. Various graphical glitches. Random crashes.
+        ppp2nd,a            Fully playable with graphical glitches. No network or DVD support. Crashes when returning to game mode from test menu.
+        p911(all)           Goes in-game. Controllers are not emulated. Various graphical glitches.
+        tsurugi,j           Goes in-game. Controllers are not emulated. Various graphical glitches.
+
+        gticlub2,ea         Attract mode works. Coins up. Hangs in various places. Will crash with "network error" after stage is selected.
+        thrild2,a           Attract mode with partial graphics. Coins up. Hangs in various places.
+
+        sscopefh            Graphics heavily glitched. Gun controller is not emulated. Sensor error and hopper error stop it from working.
+
+        mfightc,c           Requires touch panel emulation. Gets stuck at "Waiting for central monitor, checking serial...".
```

Comment:
We don't have code for the main unit, and since it requires that plus 8 terminals connected is pretty much hopeless.

Reply:
Yeah, there's a long comment about mfightc in the Viper code and that's the gist I got from it. Not worth a lot of effort due to the setup required. I only took a look at it to add an nvram for mfightcc, which I just now realized that I did submit to the email address, but forgot to add the hashes to the PR. Whoops. :)